### ISSUE TITLE
Inverted metrics InfluxDB2: storage_shard_write_sum & storage_shard_write_count 

### ISSUE THREAD

**Issue by thomasbromkapsdata** [influxdata/influxdb#25056]
__Steps to reproduce:__

1. Set up a basic influxdb scrapper toward influxdb metrics writing toward a "monitoring_influxdb" bucket by example.
2. Take a look at storage_shard_write_sum & storage_shard_write_count metrics for you "monitoring_influxdb" bucket (using id) using by example 10s aggregation and difference() to get the change
3. You can see that storage_shard_write_sum give you 1 every ten second that correspond to the number of write by the scrapper but that storage_shard_write_count give you a large number looking like a number of points.

__Expected behaviour:__
Metrics should follow the documentation and their names:
storage_shard_write_sum should give "Number of points in shard write requests" as intended by the "sum" at the end
storage_shard_write_count should give "Number of shard write requests" as intended by the "count" at the end

__Actual behaviour:__
storage_shard_write_sum give "Number of shard write requests"
storage_shard_write_count give "Number of points in shard write requests"

__Environment info:__
Different debian environments.
Influx 2.7.6 or Influx 2.6.1

__Config:__
Nothing special.

**Comment by lowjoel** [comment]
Because of this oversight: https://sourcegraph.com/github.com/influxdata/influxdb@937c819bf7a4b01a38ae306b80df53da432d87ed/-/blob/tsdb/shard.go?L265

```go
func (t twoCounterObserver) Observe(f float64) {
	t.sum.Inc()
	t.count.Add(f)
}
```

This should have incremented the count, and added the sum. But the code did the opposite.

**Comment by philjb** [comment]
@lowjoel - hi I agree the count/sum is reversed or the descriptions of the metrics are reserved. `Sum` I think should be the number of points.

Would you open a PR for your change?

**Comment by lowjoel** [comment]
Would an Influx 2 change be welcome? I'm also unsure if it requires a CLA; I'm still figuring that out on the Telegraf side of things (assuming it's the same CLA)

**Comment by philjb** [comment]
Yes - the pr is welcome. Yes there is a CLA - when you open a pr, there will be a link to the CLA. (probably the same one for telegraf, but I'm not sure.)
